File: BEAKJOON/Java/BOJ10773.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

//제로
public class BOJ10773 {
    public static void main(String[] args)throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int k = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();

        for(int i=0;i<k;i++){
            int num = Integer.parseInt(br.readLine());

            if(num==0){
                stack.pop();
            }
            else{
                stack.push(num);
            }
        }

        int sum = 0;
        for(int i : stack){
            sum += i;
        }
        System.out.println(sum);
    }
}

File: BEAKJOON/C++/BOJ10773.cpp
#include <iostream>
#include <stack>

//제로
using namespace std;

int main(){
    ios_base::sync_with_stdio(0);
	cin.tie(0); 
    cout.tie(0);

    int k,n;
    cin >> k;
    stack<int> st;

    for(int i=0;i<k;i++){
        cin >> n;
        if(n == 0){
            if(!st.empty()){
                st.pop();
            }
        } else {
            st.push(n);
        }
    }
    int sum = 0;
    while(!st.empty()){        
        sum += st.top();
        st.pop();
    }
    cout << sum << '\n';

    return 0;
}